chain-abci/src/app/slash_accounts.rs

```diff
@@ -97,17 +101,25 @@ impl<T: EnclaveProxy> ChainNodeApp<T> {
         accounts_to_slash: I,
     ) -> SlashRatio {
         let total_vote_power = self.get_total_vote_power();
+        let last_state = self
+            .last_state
+            .as_ref()
+            .expect("Last state is not present, init_chain was not called");
 
         let slashing_proportion = Milli::from_millis(
             accounts_to_slash
                 .map(|address| {
                     let validator_voting_power = Milli::new(
-                        i64::from(
-                            *self
-                                .validator_voting_power
-                                .get(&address)
-                                .expect("Voting power for a validator not found"),
-                        ) as u64,
+                        TendermintVotePower::from(
+                            get_account(
+                                &address,
+                                &last_state.last_account_root_hash,
+                                &self.accounts,
+                            )
+                            .expect("Voting power for a validator not found")
+                            .bonded,
+                        )
+                        .into(),
```

Comment:
@devashishdxt as `validator_voting_power` now contains only the active validators, the old code wouldn't find the inactive validator that should be punished -- not sure if this fix is ok

Reply:
I don't think this should affect the algorithm. The current slashing algorithm uses `validator_voting_power` to calculate `slash_ratio` (using `total_voting_power` and `current_validator_voting_power`).

Reply:
`total_voting_power` should be the total voting power of active validators?

Reply:
Yes. `total_voting_power` is the sum of voting powers of all the validators in that block.